R/run_pacta.R

```diff
@@ -149,215 +149,48 @@ run_pacta <- function(config) {
-      nrow()
-    if (available_rows > 0) {
-      generate_individual_outputs(
-        data = results_sda_total,
-        matched_prioritized = matched_prioritized,
-        output_directory = output_path_standard,
-        target_type = "sda",
-        group_id = sda_i,
-        scenario_source = scenario_source_input,
-        scenario = scenario_select,
-        region = region_select,
-        sector = sector_select,
-        start_year = start_year,
-        time_horizon = time_frame_select
-      )
-    } else {
-      next()
-    }
-  }
-  ### cement----
-  sector_select <- "cement"
-  for (sda_i in unique_groups_sda) {
-    available_rows <- results_sda_total %>%
-      dplyr::filter(
-        .data[["group_id"]] == .env[["tms_i"]],
-        .data[["scenario_source"]] == .env[["scenario_source_input"]],
-        grepl(.env[["scenario_select"]], .data$emission_factor_metric),
-        .data[["region"]] == .env[["region_select"]],
-        .data[["sector"]] == .env[["sector_select"]]
-      ) %>%
-      nrow()
-    if (available_rows > 0) {
-      generate_individual_outputs(
-        data = results_sda_total,
-        matched_prioritized = matched_prioritized,
-        output_directory = output_path_standard,
-        target_type = "sda",
-        group_id = sda_i,
-        scenario_source = scenario_source_input,
-        scenario = scenario_select,
-        region = region_select,
-        sector = sector_select,
-        start_year = start_year,
-        time_horizon = time_frame_select
-      )
-    } else {
-      next()
-    }
-  }
-  ### steel----
-  sector_select <- "steel"
-  for (sda_i in unique_groups_sda) {
-    available_rows <- results_sda_total %>%
-      dplyr::filter(
-        .data[["group_id"]] == .env[["tms_i"]],
-        .data[["scenario_source"]] == .env[["scenario_source_input"]],
-        grepl(.env[["scenario_select"]], .data$emission_factor_metric),
-        .data[["region"]] == .env[["region_select"]],
-        .data[["sector"]] == .env[["sector_select"]]
-      ) %>%
-      nrow()
-    if (available_rows > 0) {
-      generate_individual_outputs(
-        data = results_sda_total,
-        matched_prioritized = matched_prioritized,
-        output_directory = output_path_standard,
-        target_type = "sda",
-        group_id = sda_i,
-        scenario_source = scenario_source_input,
-        scenario = scenario_select,
-        region = region_select,
-        sector = sector_select,
-        start_year = start_year,
-        time_horizon = time_frame_select
-      )
-    } else {
-      next()
+  for (vals in sector_vals) {
+    sector_select <- vals[["sector"]]
+    groups <- get(paste0("unique_groups_", vals[["method"]]))
+
+    for (i in groups) {
+      available_rows <- results_tms_total %>%
+        dplyr::filter(
+          .data[["group_id"]] == .env[["i"]],
+          .data[["scenario_source"]] == .env[["scenario_source_input"]],
+          grepl(.env[["scenario_select"]], .data$metric),
```

Comment:
unfortunately this still does not work, because in the tmsr results, the variable is called `metric`, whereas in sda it is called `emission_factor_metric`. So this automatically leaves 0 rows for SDA sectors